tests/rest/client/test_rooms.py

```diff
@@ -966,6 +970,81 @@ async def user_may_join_room(
         self.helper.join(self.room3, self.user2, expect_code=403, tok=self.tok2)
 
 
+class RoomAppserviceTsParamTestCase(RoomBase):
```

Comment:
I think this should inherit directly from `HomeserverTestCase` rather than `RoomBase`? It seems to override everything in `RoomBase`?